python/tvm/relay/frontend/onnx.py

```diff
@@ -780,19 +784,36 @@ def _impl_v9(cls, inputs, attr, params):
             assert len(inputs) == 2, "Upsample op take 2 inputs, {} given".format(len(inputs))
             scales = params[inputs[1].name_hint].asnumpy()
             inputs = inputs[:1]
-        assert len(scales) == 4 and scales[0] == 1.0 and scales[1] == 1.0
+        assert scales[0] == 1.0 and scales[1] == 1.0
+        input_shape = infer_shape(inputs[0])
+        dims = len(input_shape)
+        isUpSample3D = bool(dims == 5)
```

Comment:
I'm not a big fan of defining a variable for this as just checking `dims == 5` in line is clear and concise enough. It also makes things cleaner for other Nd operator support.